cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -52,6 +55,13 @@ func (c *Commander) Start(ctx context.Context) error {
 		return nil
 	}
 
+	if len(c.doneCh) > 0 {
+		<-c.doneCh
+	}
+	if len(c.exitCh) > 0 {
+		<-c.exitCh
+	}
```

Comment:
This might not be safe in concurrent situations (e.g. the len check passes, but some other goroutine receives between the len check and the receive)

Instead, I think we could just do a non-blocking receive and ditch the `if` statement here:
```go
select {
  case <-c.exitCh:
  default:
}
```

Reply:
I am not sure if this has to be concurrent safe, If so, I think there might be other things to look into as well.

Reply:
Good point, I just assumed it was meant to be concurrent safe due to the use of atomics. 

I think the use of the `running` variable has to be re-examined at some point, because right now it's definitely not concurrent safe.